sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```
@@ -423,6 +424,100 @@ case class ShowTablePropertiesCommand(
+ * }}}
+ */
+case class ShowPartitionsCommand(
+    table: TableIdentifier,
+    partitionSpec: Option[Map[String, String]]) extends RunnableCommand {
+  // The result of SHOW PARTITIONS has one column called 'result'
+  override val output: Seq[Attribute] = {
+    AttributeReference("result", StringType, nullable = false)() :: Nil
+  }
+
+  /**
+   * This function validates the partitioning spec by making sure all the referenced columns are
+   * defined as partitioning columns in table definition. An AnalysisException exception is
+   * thrown if the partitioning spec is invalid.
+   */
+  private def validatePartitionSpec(table: CatalogTable, spec: Map[String, String]): Unit = {
+    if (!spec.keySet.forall(table.partitionColumns.map(_.name).contains)) {
+      throw new AnalysisException(s"Partition spec ${spec.mkString("(", ", ", ")")} contains " +
+        s"non-partition columns")
+    }
+  }
+
+  /**
+   * Validates and throws an [[AnalysisException]] exception under the following conditions:
+   * 1. If the table is not partitioned.
+   * 2. If it is a datasource table.
+   * 3. If it is a view or index table.
+   */
+  private def checkRequirements(table: CatalogTable): Unit = {
+    if (table.tableType == CatalogTableType.VIRTUAL_VIEW ||
+      table.tableType == CatalogTableType.INDEX_TABLE) {
+      throw new AnalysisException("Operation not allowed: view or index table")
+    } else if (!DDLUtils.isTablePartitioned(table)) {
+      throw new AnalysisException(s"Table ${table.qualifiedName} is not a partitioned table")
+    } else if (DDLUtils.isDatasourceTable(table)) {
+      throw new AnalysisException("Operation not allowed: datasource table")
+    }
+  }
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    val catalog = sqlContext.sessionState.catalog
+    val db = table.database.getOrElse(catalog.getCurrentDatabase)
+    if (catalog.isTemporaryTable(table)) {
+      Seq.empty[Row]
```

I think we should throw an exception instead if it's a temporary table